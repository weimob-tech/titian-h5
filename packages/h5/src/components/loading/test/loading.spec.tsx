import { h } from '@stencil/core';
import { newSpecPage } from '@stencil/core/testing';
import { TiLoading } from '..';
import { TestContainer } from '../../common/test/container';

describe('渲染加载中组件', () => {
  it('基本使用', async () => {
    const page = await newSpecPage({
      components: [TiLoading, TestContainer],
      template: () => (
        <test-container>
          <ti-loading />
          <ti-loading mode="circular" />
          <ti-loading mode="spinner" />
        </test-container>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });

  it('自定义加载颜色, 尺寸', async () => {
    const page = await newSpecPage({
      components: [TiLoading, TestContainer],
      template: () => (
        <test-container>
          <ti-loading color="#ff0000" mode="circular" />
          <ti-loading color="#ff0000" mode="spinner" />
          <ti-loading mode="circular" size={108} />
          <ti-loading mode="spinner" size={108} />
        </test-container>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });

  it('添加文字', async () => {
    const page = await newSpecPage({
      components: [TiLoading, TestContainer],
      template: () => (
        <test-container>
          <ti-loading mode="circular" text="loading..." />
          <ti-loading mode="spinner" text="loading..." />
          <ti-loading color="#ff0000" mode="circular" text="loading..." />
          <ti-loading color="#ff0000" mode="spinner" text="loading..." />
        </test-container>
      ),
    });
    expect(page.root).toMatchSnapshot();
  });
});
