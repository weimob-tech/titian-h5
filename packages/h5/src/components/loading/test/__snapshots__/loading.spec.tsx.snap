// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`渲染加载中组件 基本使用 1`] = `
<test-container>
  <!---->
  <ti-loading>
    <mock:shadow-root>
      <div class="titian-loading-wrap" style="flex-direction: row;">
        <div class="titian-loading titian-loading-circular" part="" style="padding: 3px; width: 30px; height: 30px;">
          <div class="titian-loading-view titian-loading-view-circular" style="width: 30px; height: 30px;">
            <div class="titian-loading-view-item" style="width: 30px; height: 30px; border-width: 2.5px;"></div>
          </div>
          <div class="titian-loading-view titian-loading-view-circular" style="width: 30px; height: 30px;">
            <div class="titian-loading-view-item" style="width: 30px; height: 30px; border-width: 2.5px;"></div>
          </div>
          <div class="titian-loading-view titian-loading-view-circular" style="width: 30px; height: 30px;">
            <div class="titian-loading-view-item" style="width: 30px; height: 30px; border-width: 2.5px;"></div>
          </div>
          <div class="titian-loading-view titian-loading-view-circular" style="width: 30px; height: 30px;">
            <div class="titian-loading-view-item" style="width: 30px; height: 30px; border-width: 2.5px;"></div>
          </div>
          <div class="titian-loading-view titian-loading-view-circular" style="width: 30px; height: 30px;">
            <div class="titian-loading-view-item" style="width: 30px; height: 30px; border-width: 2.5px;"></div>
          </div>
        </div>
        <div class="titian-loading-text titian-loading-text-empty titian-loading-text-row">
          <slot></slot>
        </div>
      </div>
    </mock:shadow-root>
  </ti-loading>
  <ti-loading>
    <mock:shadow-root>
      <div class="titian-loading-wrap" style="flex-direction: row;">
        <div class="titian-loading titian-loading-circular" part="" style="padding: 3px; width: 30px; height: 30px;">
          <div class="titian-loading-view titian-loading-view-circular" style="width: 30px; height: 30px;">
            <div class="titian-loading-view-item" style="width: 30px; height: 30px; border-width: 2.5px;"></div>
          </div>
          <div class="titian-loading-view titian-loading-view-circular" style="width: 30px; height: 30px;">
            <div class="titian-loading-view-item" style="width: 30px; height: 30px; border-width: 2.5px;"></div>
          </div>
          <div class="titian-loading-view titian-loading-view-circular" style="width: 30px; height: 30px;">
            <div class="titian-loading-view-item" style="width: 30px; height: 30px; border-width: 2.5px;"></div>
          </div>
          <div class="titian-loading-view titian-loading-view-circular" style="width: 30px; height: 30px;">
            <div class="titian-loading-view-item" style="width: 30px; height: 30px; border-width: 2.5px;"></div>
          </div>
          <div class="titian-loading-view titian-loading-view-circular" style="width: 30px; height: 30px;">
            <div class="titian-loading-view-item" style="width: 30px; height: 30px; border-width: 2.5px;"></div>
          </div>
        </div>
        <div class="titian-loading-text titian-loading-text-empty titian-loading-text-row">
          <slot></slot>
        </div>
      </div>
    </mock:shadow-root>
  </ti-loading>
  <ti-loading>
    <mock:shadow-root>
      <div class="titian-loading-wrap" style="flex-direction: row;">
        <div class="titian-loading titian-loading-spinner" part="" style="padding: 3px; width: 30px; height: 30px;">
          <div class="titian-loading-view titian-loading-view-spinner" style="width: 30px; height: 30px;"></div>
          <div class="titian-loading-view titian-loading-view-spinner" style="width: 30px; height: 30px;"></div>
          <div class="titian-loading-view titian-loading-view-spinner" style="width: 30px; height: 30px;"></div>
          <div class="titian-loading-view titian-loading-view-spinner" style="width: 30px; height: 30px;"></div>
          <div class="titian-loading-view titian-loading-view-spinner" style="width: 30px; height: 30px;"></div>
          <div class="titian-loading-view titian-loading-view-spinner" style="width: 30px; height: 30px;"></div>
          <div class="titian-loading-view titian-loading-view-spinner" style="width: 30px; height: 30px;"></div>
          <div class="titian-loading-view titian-loading-view-spinner" style="width: 30px; height: 30px;"></div>
        </div>
        <div class="titian-loading-text titian-loading-text-empty titian-loading-text-row">
          <slot></slot>
        </div>
      </div>
    </mock:shadow-root>
  </ti-loading>
</test-container>
`;

exports[`渲染加载中组件 添加文字 1`] = `
<test-container>
  <!---->
  <ti-loading>
    <mock:shadow-root>
      <div class="titian-loading-wrap" style="flex-direction: row;">
        <div class="titian-loading titian-loading-circular" part="" style="padding: 3px; width: 30px; height: 30px;">
          <div class="titian-loading-view titian-loading-view-circular" style="width: 30px; height: 30px;">
            <div class="titian-loading-view-item" style="width: 30px; height: 30px; border-width: 2.5px;"></div>
          </div>
          <div class="titian-loading-view titian-loading-view-circular" style="width: 30px; height: 30px;">
            <div class="titian-loading-view-item" style="width: 30px; height: 30px; border-width: 2.5px;"></div>
          </div>
          <div class="titian-loading-view titian-loading-view-circular" style="width: 30px; height: 30px;">
            <div class="titian-loading-view-item" style="width: 30px; height: 30px; border-width: 2.5px;"></div>
          </div>
          <div class="titian-loading-view titian-loading-view-circular" style="width: 30px; height: 30px;">
            <div class="titian-loading-view-item" style="width: 30px; height: 30px; border-width: 2.5px;"></div>
          </div>
          <div class="titian-loading-view titian-loading-view-circular" style="width: 30px; height: 30px;">
            <div class="titian-loading-view-item" style="width: 30px; height: 30px; border-width: 2.5px;"></div>
          </div>
        </div>
        <div class="titian-loading-text titian-loading-text-row">
          loading...
        </div>
      </div>
    </mock:shadow-root>
  </ti-loading>
  <ti-loading>
    <mock:shadow-root>
      <div class="titian-loading-wrap" style="flex-direction: row;">
        <div class="titian-loading titian-loading-spinner" part="" style="padding: 3px; width: 30px; height: 30px;">
          <div class="titian-loading-view titian-loading-view-spinner" style="width: 30px; height: 30px;"></div>
          <div class="titian-loading-view titian-loading-view-spinner" style="width: 30px; height: 30px;"></div>
          <div class="titian-loading-view titian-loading-view-spinner" style="width: 30px; height: 30px;"></div>
          <div class="titian-loading-view titian-loading-view-spinner" style="width: 30px; height: 30px;"></div>
          <div class="titian-loading-view titian-loading-view-spinner" style="width: 30px; height: 30px;"></div>
          <div class="titian-loading-view titian-loading-view-spinner" style="width: 30px; height: 30px;"></div>
          <div class="titian-loading-view titian-loading-view-spinner" style="width: 30px; height: 30px;"></div>
          <div class="titian-loading-view titian-loading-view-spinner" style="width: 30px; height: 30px;"></div>
        </div>
        <div class="titian-loading-text titian-loading-text-row">
          loading...
        </div>
      </div>
    </mock:shadow-root>
  </ti-loading>
  <ti-loading>
    <mock:shadow-root>
      <div class="titian-loading-wrap" style="flex-direction: row;">
        <div class="titian-loading titian-loading-circular" part="" style="padding: 3px; width: 30px; height: 30px; --loading-color: #ff0000;">
          <div class="titian-loading-view titian-loading-view-circular" style="width: 30px; height: 30px;">
            <div class="titian-loading-view-item" style="width: 30px; height: 30px; border-width: 2.5px; border-top-color: #ff0000;"></div>
          </div>
          <div class="titian-loading-view titian-loading-view-circular" style="width: 30px; height: 30px;">
            <div class="titian-loading-view-item" style="width: 30px; height: 30px; border-width: 2.5px; border-top-color: #ff0000;"></div>
          </div>
          <div class="titian-loading-view titian-loading-view-circular" style="width: 30px; height: 30px;">
            <div class="titian-loading-view-item" style="width: 30px; height: 30px; border-width: 2.5px; border-top-color: #ff0000;"></div>
          </div>
          <div class="titian-loading-view titian-loading-view-circular" style="width: 30px; height: 30px;">
            <div class="titian-loading-view-item" style="width: 30px; height: 30px; border-width: 2.5px; border-top-color: #ff0000;"></div>
          </div>
          <div class="titian-loading-view titian-loading-view-circular" style="width: 30px; height: 30px;">
            <div class="titian-loading-view-item" style="width: 30px; height: 30px; border-width: 2.5px; border-top-color: #ff0000;"></div>
          </div>
        </div>
        <div class="titian-loading-text titian-loading-text-row" style="color: #ff0000;">
          loading...
        </div>
      </div>
    </mock:shadow-root>
  </ti-loading>
  <ti-loading>
    <mock:shadow-root>
      <div class="titian-loading-wrap" style="flex-direction: row;">
        <div class="titian-loading titian-loading-spinner" part="" style="padding: 3px; width: 30px; height: 30px; --loading-color: #ff0000;">
          <div class="titian-loading-view titian-loading-view-spinner" style="width: 30px; height: 30px;"></div>
          <div class="titian-loading-view titian-loading-view-spinner" style="width: 30px; height: 30px;"></div>
          <div class="titian-loading-view titian-loading-view-spinner" style="width: 30px; height: 30px;"></div>
          <div class="titian-loading-view titian-loading-view-spinner" style="width: 30px; height: 30px;"></div>
          <div class="titian-loading-view titian-loading-view-spinner" style="width: 30px; height: 30px;"></div>
          <div class="titian-loading-view titian-loading-view-spinner" style="width: 30px; height: 30px;"></div>
          <div class="titian-loading-view titian-loading-view-spinner" style="width: 30px; height: 30px;"></div>
          <div class="titian-loading-view titian-loading-view-spinner" style="width: 30px; height: 30px;"></div>
        </div>
        <div class="titian-loading-text titian-loading-text-row" style="color: #ff0000;">
          loading...
        </div>
      </div>
    </mock:shadow-root>
  </ti-loading>
</test-container>
`;

exports[`渲染加载中组件 自定义加载颜色, 尺寸 1`] = `
<test-container>
  <!---->
  <ti-loading>
    <mock:shadow-root>
      <div class="titian-loading-wrap" style="flex-direction: row;">
        <div class="titian-loading titian-loading-circular" part="" style="padding: 3px; width: 30px; height: 30px; --loading-color: #ff0000;">
          <div class="titian-loading-view titian-loading-view-circular" style="width: 30px; height: 30px;">
            <div class="titian-loading-view-item" style="width: 30px; height: 30px; border-width: 2.5px; border-top-color: #ff0000;"></div>
          </div>
          <div class="titian-loading-view titian-loading-view-circular" style="width: 30px; height: 30px;">
            <div class="titian-loading-view-item" style="width: 30px; height: 30px; border-width: 2.5px; border-top-color: #ff0000;"></div>
          </div>
          <div class="titian-loading-view titian-loading-view-circular" style="width: 30px; height: 30px;">
            <div class="titian-loading-view-item" style="width: 30px; height: 30px; border-width: 2.5px; border-top-color: #ff0000;"></div>
          </div>
          <div class="titian-loading-view titian-loading-view-circular" style="width: 30px; height: 30px;">
            <div class="titian-loading-view-item" style="width: 30px; height: 30px; border-width: 2.5px; border-top-color: #ff0000;"></div>
          </div>
          <div class="titian-loading-view titian-loading-view-circular" style="width: 30px; height: 30px;">
            <div class="titian-loading-view-item" style="width: 30px; height: 30px; border-width: 2.5px; border-top-color: #ff0000;"></div>
          </div>
        </div>
        <div class="titian-loading-text titian-loading-text-empty titian-loading-text-row" style="color: #ff0000;">
          <slot></slot>
        </div>
      </div>
    </mock:shadow-root>
  </ti-loading>
  <ti-loading>
    <mock:shadow-root>
      <div class="titian-loading-wrap" style="flex-direction: row;">
        <div class="titian-loading titian-loading-spinner" part="" style="padding: 3px; width: 30px; height: 30px; --loading-color: #ff0000;">
          <div class="titian-loading-view titian-loading-view-spinner" style="width: 30px; height: 30px;"></div>
          <div class="titian-loading-view titian-loading-view-spinner" style="width: 30px; height: 30px;"></div>
          <div class="titian-loading-view titian-loading-view-spinner" style="width: 30px; height: 30px;"></div>
          <div class="titian-loading-view titian-loading-view-spinner" style="width: 30px; height: 30px;"></div>
          <div class="titian-loading-view titian-loading-view-spinner" style="width: 30px; height: 30px;"></div>
          <div class="titian-loading-view titian-loading-view-spinner" style="width: 30px; height: 30px;"></div>
          <div class="titian-loading-view titian-loading-view-spinner" style="width: 30px; height: 30px;"></div>
          <div class="titian-loading-view titian-loading-view-spinner" style="width: 30px; height: 30px;"></div>
        </div>
        <div class="titian-loading-text titian-loading-text-empty titian-loading-text-row" style="color: #ff0000;">
          <slot></slot>
        </div>
      </div>
    </mock:shadow-root>
  </ti-loading>
  <ti-loading>
    <mock:shadow-root>
      <div class="titian-loading-wrap" style="flex-direction: row;">
        <div class="titian-loading titian-loading-circular" part="" style="padding: 4.5px; width: 45px; height: 45px;">
          <div class="titian-loading-view titian-loading-view-circular" style="width: 45px; height: 45px;">
            <div class="titian-loading-view-item" style="width: 45px; height: 45px; border-width: 3.75px;"></div>
          </div>
          <div class="titian-loading-view titian-loading-view-circular" style="width: 45px; height: 45px;">
            <div class="titian-loading-view-item" style="width: 45px; height: 45px; border-width: 3.75px;"></div>
          </div>
          <div class="titian-loading-view titian-loading-view-circular" style="width: 45px; height: 45px;">
            <div class="titian-loading-view-item" style="width: 45px; height: 45px; border-width: 3.75px;"></div>
          </div>
          <div class="titian-loading-view titian-loading-view-circular" style="width: 45px; height: 45px;">
            <div class="titian-loading-view-item" style="width: 45px; height: 45px; border-width: 3.75px;"></div>
          </div>
          <div class="titian-loading-view titian-loading-view-circular" style="width: 45px; height: 45px;">
            <div class="titian-loading-view-item" style="width: 45px; height: 45px; border-width: 3.75px;"></div>
          </div>
        </div>
        <div class="titian-loading-text titian-loading-text-empty titian-loading-text-row">
          <slot></slot>
        </div>
      </div>
    </mock:shadow-root>
  </ti-loading>
  <ti-loading>
    <mock:shadow-root>
      <div class="titian-loading-wrap" style="flex-direction: row;">
        <div class="titian-loading titian-loading-spinner" part="" style="padding: 4.5px; width: 45px; height: 45px;">
          <div class="titian-loading-view titian-loading-view-spinner" style="width: 45px; height: 45px;"></div>
          <div class="titian-loading-view titian-loading-view-spinner" style="width: 45px; height: 45px;"></div>
          <div class="titian-loading-view titian-loading-view-spinner" style="width: 45px; height: 45px;"></div>
          <div class="titian-loading-view titian-loading-view-spinner" style="width: 45px; height: 45px;"></div>
          <div class="titian-loading-view titian-loading-view-spinner" style="width: 45px; height: 45px;"></div>
          <div class="titian-loading-view titian-loading-view-spinner" style="width: 45px; height: 45px;"></div>
          <div class="titian-loading-view titian-loading-view-spinner" style="width: 45px; height: 45px;"></div>
          <div class="titian-loading-view titian-loading-view-spinner" style="width: 45px; height: 45px;"></div>
        </div>
        <div class="titian-loading-text titian-loading-text-empty titian-loading-text-row">
          <slot></slot>
        </div>
      </div>
    </mock:shadow-root>
  </ti-loading>
</test-container>
`;
